src/tdycoremesh.c

```diff
@@ -0,0 +1,1230 @@
+#include <petsc.h>
+#include <private/tdycoreimpl.h>
+#include <private/tdymeshimpl.h>
+#include <private/tdymemoryimpl.h>
+#include <private/tdyutils.h>
+
+/* ---------------------------------------------------------------- */
+
+PetscBool IsClosureWithinBounds(PetscInt closure, PetscInt start,
+                                PetscInt end) {
+  return (closure >= start) && (closure < end);
+}
+
+/* -------------------------------------------------------------------------- */
+PetscErrorCode AllocateMemoryForASubcell(
+  TDy_subcell    *subcell,
+  TDySubcellType subcell_type) {
+
+  PetscFunctionBegin;
+
+  PetscInt num_nu_vectors;
```

Comment:
Because these get set in the select statement, the compiler complains that they might not be initialized. Can we initially set them to 0 to squelch this warning?

Reply:
I instead add an error message under the `default` case within the `switch(subcell_type)` https://github.com/TDycores-Project/TDycore/pull/5/commits/e7441c7e94d399ab7f76cee3617631efa9a1ae3a#diff-fcf52e8a4105ab2e05e6a4f90c5dfbd9R33